src/coreclr/vm/jitinterface.cpp

```diff
@@ -1830,44 +1911,24 @@ void CEEInfo::getThreadLocalStaticBlocksInfo (CORINFO_THREAD_STATIC_BLOCKS_INFO*
         pInfo->offsetOfThreadStaticBlocks = CEEInfo::ThreadLocalOffset(&t_NonGCThreadStaticBlocks);
         pInfo->offsetOfMaxThreadStaticBlocks = CEEInfo::ThreadLocalOffset(&t_NonGCMaxThreadStaticBlocks);
```

Comment:
Move the whole `if (isGCType) { ... } else { ... }` outside the platform specific ifdef, and implement `CEEInfo::ThreadLocalOffset` by subtracting address of the thread static from address returned by calling `__tls_get_add(getModuleThreadStaticDescriptor())`.

Reply:
Alternatively, you can wrap the 3 thread_local statics in a struct so that the relative offset are known, make the helper return the address of the struct (`keep `TLSGD`), and return the relative offsets in the structs. I think the alternative may be tiny bit better since one of the offsets is going to 0 that will produce smaller instruction on x64.

Note that the JIT/EE interface would be the same for both of these approaches.

Reply:
> Alternatively, you can wrap the 3 thread_local statics in a struct so that the relative offset are known

Done.